static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx

```diff
@@ -1,126 +1,178 @@
-import {useState} from 'react';
 import styled from '@emotion/styled';
 
-import replaysInlineOnboarding from 'sentry-images/spot/replay-inline-onboarding.svg';
+import replayInlineOnboarding from 'sentry-images/spot/replay-inline-onboarding-v2.svg';
 
 import {Button} from 'sentry/components/button';
-import ButtonBar from 'sentry/components/buttonBar';
+import {DropdownMenu} from 'sentry/components/dropdownMenu';
 import {EventReplaySection} from 'sentry/components/events/eventReplay/eventReplaySection';
-import {t} from 'sentry/locale';
+import HookOrDefault from 'sentry/components/hookOrDefault';
+import platforms, {otherPlatform} from 'sentry/data/platforms';
+import {IconClose} from 'sentry/icons';
+import {t, tct} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
-import localStorage from 'sentry/utils/localStorage';
+import type {PlatformKey} from 'sentry/types';
+import {trackAnalytics} from 'sentry/utils/analytics';
 import {useReplayOnboardingSidebarPanel} from 'sentry/utils/replays/hooks/useReplayOnboarding';
+import theme from 'sentry/utils/theme';
+import useDismissAlert from 'sentry/utils/useDismissAlert';
+import useMedia from 'sentry/utils/useMedia';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type OnboardingCTAProps = {
+  platform: PlatformKey;
+  projectId: string;
+};
+
+const OnboardingCTAButton = HookOrDefault({
+  hookName: 'component:replay-onboarding-cta-button',
+  defaultComponent: null,
+});
+
+export default function ReplayInlineOnboardingPanel({
+  platform,
+  projectId,
+}: OnboardingCTAProps) {
+  const LOCAL_STORAGE_KEY = `${projectId}:issue-details-replay-onboarding-hide`;
+
+  const {dismiss: snooze, isDismissed: isSnoozed} = useDismissAlert({
+    key: LOCAL_STORAGE_KEY,
+    expirationDays: 7,
+  });
 
-const LOCAL_STORAGE_KEY = 'replay-preview-onboarding-hide-until';
-const SNOOZE_TIME = 1000 * 60 * 60 * 24 * 7; // 1 week
-const DISMISS_TIME = 1000 * 60 * 60 * 24 * 365; // 1 year
-
-function getHideUntilTime() {
-  return Number(localStorage.getItem(LOCAL_STORAGE_KEY)) || 0;
-}
-
-function setHideUntilTime(offset: number) {
-  localStorage.setItem(LOCAL_STORAGE_KEY, String(Date.now() + offset));
-}
-
-function clearHideUntilTime() {
-  localStorage.removeItem(LOCAL_STORAGE_KEY);
-}
-
-export default function ReplayInlineOnboardingPanel() {
-  const [isHidden, setIsHidden] = useState(() => {
-    const hideUntilTime = getHideUntilTime();
-    if (hideUntilTime && Date.now() < hideUntilTime) {
-      return true;
-    }
-    clearHideUntilTime();
-    return false;
+  const {dismiss, isDismissed} = useDismissAlert({
+    key: LOCAL_STORAGE_KEY,
+    expirationDays: 365,
```

Comment:
this might make more sense as a guide

Reply:
@scttcper what do you mean by guide?

Reply:
[these](https://github.com/getsentry/sentry/blob/93560a1b9d01c5bd3e5628ef498daf2205b90198/src/sentry/utils/prompts.py#L4-L19), we can store something in the database so the banner doesn't come back when the browser clears localstorage

Reply:
are there any specific benefits to using this vs just local storage? the original replay CTA was using local storage as well

Reply:
it persists for longer and across browsers are the only advantages

Reply:
cc @vuluongj20 @Jesse-Box @rachrwang @jas-kas @ryan953 @bruno-garcia

Reply:
@scttcper how strongly do you feel about this on a [logaf](https://blog.danlew.net/2020/04/15/the-logaf-scale/) scale?

Reply:
idk how @scttcper feels, but we can always update the implementation later.

Right now we're benefitting from a new design which i'd like to see in the app.

Reply:
synced w/ @rachrwang & @bruno-garcia and they don't feel too strongly about it either; might not be worth the extra effort